Signal/src/ViewControllers/MessageDetailViewController.swift

```diff
@@ -567,4 +589,93 @@ class MessageDetailViewController: OWSViewController {
                               comment: "Status label for messages which are failed.")
         }
     }
+
+    // MARK: - Text Layout
+
+    // UITextView can't render extremely long text due to constraints on the size
+    // of its backing buffer, especially when we're embedding it "full-size' 
+    // within a UIScrollView as we do in this view.  Therefore if we do the naive
+    // thing and embed a full-size UITextView inside our UIScrollView, it will 
+    // fail to render any text if the text message is sufficiently long.
+    //
+    // Therefore we're doing something unusual.  
+    //
+    // * We use an empty UIView "messageTextProxyView" as a placeholder for the
+    //   the UITextView.  It has the size and position of where the UITextView
+    //   would be normally.
+    // * We use a UITextView inside that proxy that is just large enough to
+    //   render the content onscreen. We then move it around within the proxy
+    //   bounds to render the parts of the proxy which are onscreen.
+    private func updateTextLayout() {
```

Comment:
I've tested and this works well on iPhone 5C and can handle very large oversize text messages.  It's pretty smooth visually, too. I was afraid there might be some rendering lag.

Reply:
This is wild. At what file size does this become relevant?

Reply:
The issue shows up quite easily if you paste oversize text.  Neither `UILabel` and `UITextView` are designed to be used inside a `UIScrollView` to render long strings.

Reply:
Incidentally, I didn't come up with this solution, I've seen it used before to work around other forms of the `UIScrollview`-inside-a-`UIScrollview` issue.